riscv/triggers.h

```diff
@@ -58,6 +58,7 @@ class trigger_t {
 
   virtual match_result_t memory_access_match(processor_t UNUSED * const proc,
       operation_t UNUSED operation, reg_t UNUSED address, std::optional<reg_t> UNUSED data) { return MATCH_NONE; }
+  virtual match_result_t trap_taking_match(processor_t UNUSED * const proc, trap_t UNUSED & t) { return MATCH_NONE; }
```

Comment:
I don't understand the name of this method. Method names should be verbs. What does `trap_taking_match` do? It sounds like it should throw a trap, but it doesn't.

Reply:
How about trap_match?

Reply:
I choose ```detect_trap_match``` and add a refactor commit for ```detect_memory_access_match```.